airbyte-integrations/connectors/source-gitlab/source_gitlab/streams.py

```diff
@@ -0,0 +1,338 @@
+        yield from super().read_records(**kwargs)
+
+    def request_params(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+        params = {"page": self.page, "per_page": self.per_page}
+        if next_page_token:
+            params.update(next_page_token)
+        params.update(self.stream_base_params)
+        return params
+
+    @property
+    def url_base(self) -> str:
+        return f"https://{self.api_url}//api/v4/"
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        response_data = response.json()
+        if isinstance(response_data, dict):
+            return None
+        if len(response_data) == self.per_page:
+            self.page += 1
+            return {"page": self.page}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        response_data = response.json()
+        if isinstance(response_data, list):
+            for record in response_data:
+                yield self.transform(record, **kwargs)
+        elif isinstance(response_data, dict):
+            yield self.transform(response_data, **kwargs)
+        else:
+            Exception(f"Unsupported type of response data for stream {self.name}")
+
+    def transform(self, record: Dict[str, Any], stream_slice: Mapping[str, Any] = None, **kwargs):
+        for key in self.flatten_id_keys:
+            self._flatten_id(record, key)
+
+        for key in self.flatten_list_keys:
+            self._flatten_list(record, key)
+
+        return record
+
+    def _flatten_id(self, record: Dict[str, Any], target: str):
+        if record.get(target):
+            record[target + "_id"] = record.pop(target, {}).pop("id", None)
+        elif target in record:
+            record[target + "_id"] = record.pop(target)
+        else:
+            record[target + "_id"] = None
+
+    def _flatten_list(self, record: Dict[str, Any], target: str):
+        record[target] = [target_data.get("id") for target_data in record.get(target, [])]
+
+
+class GitlabChildStream(GitlabStream):
+    path_list = ["id"]
+
+    def __init__(self, parent_stream: GitlabStream, parent_similar: bool = False, repository_part: bool = False, **kwargs):
+        super().__init__(**kwargs)
+        self.parent_stream = parent_stream
+        self.parent_similar = parent_similar
+        self.repo_url = repository_part
+
+    @property
+    def path_template(self) -> str:
+        template = [self.parent_stream.name] + ["{" + p + "}" for p in self.path_list]
+        if self.repo_url:
+            template.append("repository")
+        template.append(casing.camel_to_snake(self.__class__.__name__))
+        return "/".join(template)
+
+    @property
+    def name(self) -> str:
+        if self.parent_similar:
+            return f"{self.parent_stream.name[:-1]}_{super().name}"
```

Comment:
Does name[:-1] for not using plural in parent stream name?

Reply:
No, all parent streams use plural